tests/functional/generate_pages.py

```diff
@@ -235,13 +241,13 @@ def find_kv(block, key, values):
     return False
 
 
-def process_block(block, dir_out, schema_data, spec_file, relative_require = '..'):
-    logger.debug("Processing Block: %s", block['id'])
+def process_block(block, dir_out, schema_data, required_spec_file, relative_require='..'):
```

Comment:
Why has `spec_file` been renamed `required_spec_file`? It's confusing as it's not required (it only does something with it if it's passed in).

Reply:
As the linting picked up on this and threw the below error: 
'"Shadows name 'spec file' from outer scope"'
So the name change got around it but maybe it was the